broker/neb/src/set_log_data.cc

```diff
@@ -46,6 +48,28 @@ static char* log_extract(char** lasts) {
   return data;
 }
 
+/**
+ * @brief Get the id of a log status.
+ *
+ * @param status A string corresponding to the state of the resource.
+ *
+ * @return A status code.
+ */
+static int status_id(const absl::string_view& status) {
+  int retval;
+  if (status == "DOWN" || status == "WARNING")
+    retval = 1;
+  else if (status == "UNREACHABLE" || status == "CRITICAL")
+    retval = 2;
+  else if (status == "UNKNOWN")
+    retval = 3;
+  else if (status == "PENDING")
+    retval = 4;
+  else
+    retval = 0;
+  return retval;
+}
+
 /**
  *  Get the id of a log status.
  */
```

Comment:
why not use the new version of status_id since string_view(const char *) exists?

Reply:
I was wondering the same question... So, let's go for it